CHANGELOG.md

```diff
@@ -6,454 +6,494 @@ All notable changes to this project will be documented in this file.
-    - Change labels in middleware topology
-    - Added "Server State" into Middleware Server Details
-    - Enabled search for Middleware entities
-    - Users can add Middleware Datasources and JDBC Drivers
-    - Metrics for JMS Topics and Queues
-  - Kubernetes: Cross-linking with OpenStack instances
-  - Microsoft Cloud (Azure)
-    - Handle new events: Create Security Group, VM Capture
-    - Provider-specific logging
-    - Added memory and disk utilization metrics
-    - Support floating IPs during provisioning
-    - Load Balancer inventory collection for Azure
-    - Pagination support in armrest gem
-  - Microsoft Infrastructure (SCVMM): Set CPU sockets and cores per socket
-  - Networking
-    - Allow port ranges for Load Balancers
-    - Load Balancer user interface
-    - Separate Google Network Manager
-    - NFV: VNFD Templates and VNF Stacks
-    - Nuage: Inventory of Managed Cloud Subnets
-    -  Nuage policy groups added
+    - Added image tag names for Containers in vim performance state.
+    - Added a 'fixed_compute_metric' column to chargeback
+    - Added rate assigning by tags to container images
+    - Chargeback vm group by tag
+  - Notifications
+    - Dynamic substitution in notification messages
+    - Generate for Lifecycle events
+- Database Maintenance
+  - Hourly reindex: High Churn Tables
+  - Periodic full vacuum
+  - Configure in appliance console
+- Notification Backend
+  - Model for asynchronous notifications
+  - Authentication token generation for web sockets
+  - API for notification drawer
+- PostgreSQL High Availability
+  - Added [repmgr](http://repmgr.org/)  to support automatic failover
+    - Maintain list of active standby database servers
+    - Added [pg-dsn_parser](https://github.com/ManageIQ/pg-dsn_parser) for converting DSN to a hash
+  - DB Cluster - Primary, Standbys
+  - Uses [repmgr](http://www.repmgr.org/) (replication)
+  - Failover
+    - Primary to Standby
+    - Appliance connects to new primary DB
+    - Primary/Standby DB config in Appliance Console
+    - Database-only appliance config in Appliance Console
+    - Failover Monitor
+- Tenancy
+  - Post refresh hook on OpenStack provider
+  - Create provider base tenant under a root tenant
+  - Cloud Tenant tree generated under provider base tenant
+  - Create / Update / Delete
+  - PostgreSQL High Availability
+  - Groundwork in preparation for supporting multiple entitlements
+  - ApplicationHelper#role_allows and User#role_allows? combined and moved to RBAC
+  - Post refresh hook to queue mapping of Cloud Tenants
+  - Added parent_id to CloudTenant as prerequisite for mapping OpenStack tenants to ManageIQ tenants
+  - Mapping Cloud Tenants to ManageIQ Tenants
+    - Prevent deleting mapped tenants from cloud provider
+    - Added checkbox "Tenant Mapping Enabled" to Openstack Manager
+  - Ad hoc sharing of resources across tenants
+    - Backend modeling completed
+    - Implementation in progress
```

Comment:
No need for WIP in release changelog (i.e. it's not going to be in release of Euwe).  In fact, we may want to drop the fact that we got the modeling in, because a user can't use it anyway.